src/sota_tools/CMakeLists.txt

```diff
@@ -11,19 +10,100 @@ set (GARAGE_PUSH_SRCS
   ostree_object.cc
   request_pool.cc
   treehub_server.cc
+  deploy.cc
+)
+
+add_library(sota_tools_static_lib STATIC ${SOTA_TOOLS_LIB_SRC})
+
+##### garage-push targets
+
+set (GARAGE_PUSH_SRCS
+  garage_push.cc
 )
 
+add_executable(garage-push ${GARAGE_PUSH_SRCS})
 
-add_executable(garage-push main.cc)
-add_library(garage_push_static_lib STATIC ${GARAGE_PUSH_SRCS})
-target_link_libraries(garage-push garage_push_static_lib aktualizr_static_lib
+# Export compile_commands.json for clang-check
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+target_compile_options(sota_tools_static_lib PUBLIC -std=c++11)
+
+target_include_directories(sota_tools_static_lib INTERFACE ${PROJECT_SOURCE_DIR} ${GLIB2_INCLUDE_DIRS})
+
+target_link_libraries(garage-push sota_tools_static_lib aktualizr_static_lib
   ${CURL_LIBRARIES}
   ${GLIB2_LIBRARIES}
   ${Boost_LIBRARIES}
   ${CMAKE_THREAD_LIBS_INIT}
   ${LibArchive_LIBRARIES}
   gcov)
 
+
+install(TARGETS garage-push RUNTIME DESTINATION bin)
+
+
+##### garage-deploy targets
+
+set (GARAGE_DEPLOY_SRCS
+  garage_deploy.cc
+)
+
+add_executable(garage-deploy ${GARAGE_DEPLOY_SRCS})
+target_link_libraries(garage-deploy sota_tools_static_lib aktualizr_static_lib
+  ${Boost_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
+  ${LibArchive_LIBRARIES}
+  ${LIBOSTREE_LIBRARIES}
+  ${CURL_LIBRARIES}
+  gcov)
+
+if(GARAGE_DEPLOY_VERSION)
+    add_definitions(-DGARAGE_DEPLOY_VERSION="${GARAGE_DEPLOY_VERSION}")
+else(GARAGE_DEPLOY_VERSION)
+    add_definitions(-DGARAGE_DEPLOY_VERSION="dev")
+endif(GARAGE_DEPLOY_VERSION)
+
+
+aktualizr_source_file_checks(${GARAGE_DEPLOY_SRCS})
+
+add_dependencies(build_tests garage-deploy)
+
+
+install(TARGETS garage-deploy RUNTIME DESTINATION bin COMPONENT garage_deploy)
+add_custom_target(download_garage_sign COMMAND wget https://ats-tuf-cli-releases.s3-eu-central-1.amazonaws.com/cli-0.1.0-175-g9618465.tgz -O ${CMAKE_CURRENT_BINARY_DIR}/garage-sign.tgz
+COMMAND tar -xzf ${CMAKE_CURRENT_BINARY_DIR}/garage-sign.tgz -C ${CMAKE_CURRENT_BINARY_DIR}
+COMMAND chmod +x ${CMAKE_CURRENT_BINARY_DIR}/garage-sign/bin/*)
+FILE(GLOB garage_sign_bin_files ${CMAKE_CURRENT_BINARY_DIR}/garage-sign/bin/*)
+FILE(GLOB garage_sign_lib_files ${CMAKE_CURRENT_BINARY_DIR}/garage-sign/lib/*)
+
+install(PROGRAMS ${garage_sign_bin_files} DESTINATION /usr/bin COMPONENT garage_deploy)
+install(FILES ${garage_sign_lib_files} DESTINATION /usr/lib COMPONENT garage_deploy)
```

Comment:
These should be use `RUNTIME DESTINATION bin` and `lib` instead of hardcoding `/usr/bin` and `/usr/lib`. See https://github.com/advancedtelematic/aktualizr/pull/384/commits/a0c50801d1a9472ccbf7c8d73d9032529913ac31.